### PR TITLE
Add properties to answers

### PR DESCRIPTION
**Pull request by adammahmood**
### What is the context of this PR?
This is part of a wider piece of work to introduce answer properties to help the quality of the respondent's input.  

Prior to this work, the concept of answer properties was limited to capturing whether or not an answer is 'mandatory'.  This concept has been refactored to support multiple properties and this PR specifically introduces 'decimal' properties.

[Trello Ticket](https://trello.com/c/f978JRq6/402-properties-decimals-1) 

This PR works in conjunction with the following PR's:

Web App - https://github.com/ONSdigital/eq-author/pull/376
Schema - ONSdigital/eq-author-graphql-schema#46

### How to review 
- Run Tests

### CODE REVIEW

**Comment by adammahmood** [comment]
This will require updates to the schema and appropriate version numbers being added to package.json

**Comment by adammahmood** [comment]
[Properties - decimals (1)](https://trello.com/c/f978JRq6/402-properties-decimals-1)

File: schema/resolvers.js [hunk 1]
```diff
@@ -92,7 +102,9 @@ const Resolvers = {
       ctx.repositories.QuestionPage.undelete(args.input.id),
 
     createAnswer: async (root, args, ctx) => {
-      const answer = await ctx.repositories.Answer.insert(args.input);
+      const defaultProperties = getDefaultAnswerProperties(args.input.type);
+      const input = merge({}, args.input, { properties: defaultProperties });
```
Comment (samiwel) [inline schema/resolvers.js]: I think we can get away with not setting default properties on Answer when it's created. When resolving Answer a field resolver for `properties` could do the merge at that point of the default properties and any overridden property values stored on the Answers table.

Comment (samiwel) [reply schema/resolvers.js]: On the other hand, setting properties on creation doesn't lead to unexpected consequences when updating defaults, plus the properties are going to be queried more often than they are created so perhaps makes more sense to leave as is.

File: schema/resolvers.js [hunk 2]
```diff
@@ -16,6 +17,15 @@ const assertMultipleChoiceAnswer = answer => {
   }
 };
 
+const getDefaultAnswerProperties = type => {
+  switch (type) {
+    case "Number":
+      return { required: true, decimals: 0 };
```
Comment (samiwel) [inline schema/resolvers.js]: According to the scenarios, currency should also get a `decimals` property defaulting to zero.

Comment (adammahmood) [reply schema/resolvers.js]: I thought we wanted to do that task separately?

Comment (samiwel) [reply schema/resolvers.js]: @adammahmood Yeah my bad, I probably should have made it explicit on the card. The task is to add a `decimals` property and according to Jimi's scenarios that applies to both `Number` and `Currency` answer types.

AFAIK this is the only place that would need to be changed to introduce it for Currency also? As it's such a small change, I think it makes sense to include it in the same PR.

Comment (adammahmood) [reply schema/resolvers.js]: Yes and probably best to write some cypress tests to validate on the UI.

File: schema/resolvers.js [hunk 3]
```diff
@@ -16,6 +17,15 @@ const assertMultipleChoiceAnswer = answer => {
...
+    default:
+      return { required: true };
```
Comment (SamGodwin2) [inline schema/resolvers.js]: How come we've switched the default case to mandatory/required being true instead of it being false like before?

**Comment by samiwel** [comment]
LGTM, this just needs to be updated to point at schema v0.16.0 once it's been published.

**Pull request merged**